crates/router/src/utils.rs

```diff
@@ -150,3 +150,11 @@ pub fn to_currency_base_unit_asf64(
     };
     Ok(amount)
 }
+
+#[inline]
+pub fn get_payment_attempt_id(
+    payment_id: impl AsRef<str> + std::fmt::Display,
```

Comment:
This would allow anything that implements `Display` to be passed? Would that be the intended behavior though?

Reply:
`impl AsRef<str> + std::fmt::Display` also would allow any string to be passed. 

In case of `impl std::fmt::Display,` , in future if payment_id is made a domain type like `PaymentId`, then implementing Display over it would suffice. And the function is just using the display implementation so I think adding just `Display` would suffice.